Listings redesign [MAILPOET-2780] [MAILPOET-2782]

NeosinneR opened the pull request:
[MAILPOET-2780]
[MAILPOET-2782]

[MAILPOET-2780]: https://mailpoet.atlassian.net/browse/MAILPOET-2780
[MAILPOET-2782]: https://mailpoet.atlassian.net/browse/MAILPOET-2782

NeosinneR commented:
@wxa this is a rather big PR, but it couldn't be easily split into several PRs. The mobile version is not exactly according to design, because the table layout can't be that easily reordered and from a maintenance point, it's not worth doing it. If you have any questions about why I did something the way I did, you can ping me on Slack.

wxa reviewed (changes requested):
Overall looks fine, found only a few issues:

1. Action buttons in mobile version have bigger margins in the design;
2. The MailPoet menu in the left admin bar is not open when in the newsletter editor;
3. There are a few visual issues with notices and screen options:
3.1. They overlap on the newsletter page:
![mailpoet-listings-redesign-notice-overlap](https://user-images.githubusercontent.com/9365088/95992217-6591aa00-0e36-11eb-98c6-beb4ce2010d4.png)
3.2. No margin between subscriber counts and a notice:
![mailpoet-listings-redesign-sub-count-margin](https://user-images.githubusercontent.com/9365088/95992333-83f7a580-0e36-11eb-95f2-94cf91894840.png)
3.3. In welcome email, post notification, woocommerce tabs screen options are misaligned:
![mailpoet-listings-redesign-screen-options-misaligned](https://user-images.githubusercontent.com/9365088/95992430-9bcf2980-0e36-11eb-87f2-9de1578e0256.png)
3.4. Should we redesign the open screen options panel? Change the button color, for example:
![image](https://user-images.githubusercontent.com/9365088/95993147-6a0a9280-0e37-11eb-82f1-b4398117e085.png)
4. "Back to Post Notifications" button in post notification history not redesigned;
5. "Create Segment" button in Statistics > Subscriber Engagement not redesigned.

NeosinneR commented on assets/js/src/common/listings/newsletter_status.tsx:
```diff
@@ -31,9 +37,12 @@ const NewsletterStatus = ({ scheduledFor, processed, total }: NewsletterStatusPr
     percentage = 100 * (processed / total);
   } else if (sent) {
-    label = `${total}`;
+    label = `${total} / ${total}`;
```
That was a slightly different issue. We said `Sent` which wasn't true in case of failed sending. However, we did _process_ 100% of those emails, which is what the progress indicates. 

I did this change because without it, there would just be a circle with 1 number, which looked a bit confusing ("what this means?"). But when added `X out of Y` it looks more understandable.

NeosinneR commented:
@wxa 
1. Margins are smaller so on Lists listing both buttons fit in there.

2. I can't replicate this, any idea how to reproduce? 
<img width="1059" alt="Screenshot 2020-10-15 at 14 03 56" src="https://user-images.githubusercontent.com/470616/96120767-57ec2b00-0eef-11eb-8297-a01d51e246ed.png">

3.1, 3.2, 3.3. Fixed

3.4. I was thinking about that, but decided to do the minimum required overriding of WordPress styles. Fewer things to maintain on each WordPress update.

4. Fixed all back buttons (skipped the one in Email stats, because Pavel is redesigning it and it would only create a conflict)

5. Fixed in premium PR

NeosinneR commented on assets/js/src/listing/listing.jsx:
```diff
@@ -550,10 +550,8 @@ class Listing extends React.Component {
       'mailpoet-listing-table',
       'wp-list-table',
       'widefat',
```
Good point, I removed both classes and updated our styles, so the design stays the same. Tables in the Help section remain in WordPress styles, we might get to those later, they are not that urgent.

wxa commented:
> I can't replicate this, any idea how to reproduce?
I can replicate it on `master` as well, so not related to your branch. Will deal with it separately.

Aside that everything's good!

veljkho reviewed (changes requested):
I see some UI issues:
- When there are plenty of lists, I see [the status value is too narrow](https://i.imgur.com/UivwtJ1.png)
- Here is [missing padding-bottom](https://i.imgur.com/NV4y9Cz.png), and I believe this is present also on prod.
- Post notifications Mobile: I see "[Not sent yet](https://i.imgur.com/W7aMWl4.png)" inappropriately.
- Post notifications Mobile: I see the [View History](https://i.imgur.com/W7aMWl4.png) button, I believe the best would be to make it the same size/style as the Preview button?
- Suggestion: [These values have very small](https://i.imgur.com/19s6cal.png) font size, increasing by 1px would be very comfortable to read
- [The title takes too long](https://i.imgur.com/Xcz7Pl2.png) before going to a 2nd row. The suggestion is to reduce it by 50%.

NeosinneR commented:
@veljkho I think everything is fixed

veljkho reviewed (changes requested):
Here are some issues present again/new:
- I see some [vertical line](https://i.imgur.com/m1l937y.png) at the end of the Restore link
- Issue with [a very long title](https://i.imgur.com/UesPSix.png) without going into another row is still present
- Is it okay to have a success/error message [merged with tabs](https://i.imgur.com/TPh1Qjf.png)?

NeosinneR commented:
@veljkho 1. and 3. fixed, in 2. I made the column a bit smaller, but since this is an HTML table and the column width is controlled by the browser (and content inside the cell), I have limited options in limiting the width. But I think this is good enough because other columns look good.
